### PR TITLE
Task1 features

### PR DESCRIPTION
**Pull request by mnjaggah**
Print methods

### CODE REVIEW

File: app/dojo.py [hunk 1]
```diff
@@ -133,6 +134,73 @@ def allocate_available_livingspace(self, new_person):
...
+		output = ""
+
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
+			if room.occupants:
+				output = ("\n\nROOM NAME: {} \tTYPE: {} " .format(room.name, room.room_type))
+				output += ("\n" + "-" * 40 + "\n")
+				for occupant in room.occupants:
+					output += (occupant.name + "-" + occupant.role + ", ")
+
+
```
Comment (kevintumbo) [inline app/dojo.py]: what about if there are no allocations?  wouldn't it be better if there was a message to highlight that?

**Comment by kevintumbo** [comment]
I like the use of itertools to chain the list. 
code looks ok.

File: app/dojo.py (hunk 1 above)
Comment (mnjaggah) [reply app/dojo.py]: Yes, i should definitely include that

File: app/dojo.py [hunk 2]
```diff
@@ -133,6 +134,73 @@ def allocate_available_livingspace(self, new_person):
...
+		output = ""
+
+		if self.officespace_waitinglist or self.livingspace_waitinglist:
+			output = "\n\n LIST OF ALL UNALLOCATED STAFF AND FELLOWS\n" + "*" * 50 + "\n"
+			for person in itertools.chain(self.officespace_waitinglist, self.livingspace_waitinglist):
+				output += (person.name + " " + person.email + " " + person.role + " " + person.wants_accomodation)
+
+		if filename:
+			print ("Saving unallocations list to file...")
```
Comment (Gchesta) [inline app/dojo.py]: There is a better alternative to this by using `with open (filename + ".txt", "w+") as txt_file: txt_file.write(output)`

No need to close the file as it does that itself

File: app/dojo.py [hunk 3]
```diff
@@ -133,6 +134,73 @@ def allocate_available_livingspace(self, new_person):
...
+		that have not been allocated any office or living space.
+
+		"""
+
+		output = ""
+
+		if self.officespace_waitinglist or self.livingspace_waitinglist:
+			output = "\n\n LIST OF ALL UNALLOCATED STAFF AND FELLOWS\n" + "*" * 50 + "\n"
+			for person in itertools.chain(self.officespace_waitinglist, self.livingspace_waitinglist):
+				output += (person.name + " " + person.email + " " + person.role + " " + person.wants_accomodation)
```
Comment (Gchesta) [inline app/dojo.py]: Just learnt another cool feature. will study more on this

File: app/dojo.py [hunk 4]
```diff
@@ -156,8 +156,29 @@ def print_room(self, room_name):
...
 
 	def print_allocations(self,filename=None):
+		"""This method prints a list of all allocations at the Dojo. The registered
+		allocations can then be written to the specified text file
+
+		"""
+		output = ''
 
-		pass
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
```
Comment (oscamaina) [inline app/dojo.py]: I like this. this is more efficient. learned something new

**Pull request merged**